File: Programmers/Level 2/기능개발.js
function solution(progresses, speeds) {
    let answer = []
    let done = []
    let flag = true
    while (progresses.length) {
        done = []
        for (let i=0; i<progresses.length; i++) {
            flag = true
            progresses[i] += speeds[i]
            if (progresses[i] >= 100) {
                for (let j=0; j<i; j++) {
                    if (progresses[j] < 100) {
                        flag = false
                    }
                }
                if (flag) done.push(i)
            }
        }
        if (done.length) {
            answer.push(done.length)
            while (done.length) {
                progresses.shift()
                speeds.shift()  // 이 부분 빼먹어서 오래걸림
                done.shift()
            }
        }
    }
    return answer
}

File: Programmers/Level 2/기능개발.py
def solution2(progresses, speeds):
	answer = []
	front = 0
	while front < len(progresses):
		for i in range(front, len(progresses)):
			progresses[i] += speeds[i]

		done = 0
		while front < len(progresses):
			if progresses[front] >= 100:
				front += 1
				done += 1
			else:
				break
		if done > 0:
			answer += [done]

	return answer


def solution(progresses, speeds):
	answer = []
	front = 0
	day = 1
	done = 0
	while front < len(progresses):
		if progresses[front] + speeds[front] * day >= 100:
			done += 1
			front += 1
			continue
		else:
			answer += [done]
			day += 1
			done = 0
	answer += [done]
	return list(filter(lambda x: x > 0, answer))


# print(solution([93, 30, 55], [1, 30, 5]))
# print(solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1]))
# print(solution([95, 95, 95, 95], [4, 3, 2, 1]))
